packages/components/carousel/__tests__/carousel.test.tsx

```diff
@@ -221,4 +221,41 @@ describe('Carousel', () => {
       expect(indicators[index].element.textContent).toEqual(value.toString())
     })
   })
+  it('height is set to auto', async () => {
+    const data = [1, 2, 3]
+
+    wrapper = mount({
+      setup() {
+        return () => (
+          <div>
+            <Carousel height={'auto'} autoplay={false}>
+              {data.map((value) => (
+                <CarouselItem label={value} key={value} style="height: 100px">
+                  {value}
+                </CarouselItem>
+              ))}
+            </Carousel>
+          </div>
+        )
+      },
+    })
+
+    const items = wrapper.vm.$el.querySelectorAll('.el-carousel__item')
+
+    const mockCarouselItemOffsetHeight = vi
+      .spyOn(items[0], 'offsetHeight', 'get')
+      .mockImplementation(() => 100)
+
```

Comment:
Maybe you can mock like this and get the height value dynamically
```ts
Array.from<HTMLElement>(items).forEach((item) => {
  vi.spyOn(item, 'offsetHeight', 'get').mockImplementation(() => {
    return parseFloat(window.getComputedStyle(item).height) || 0
  })
})
```

Reply:
ok, thank you, good idea, I have resubmitted the code